crates/git_ui/src/git_panel.rs

```diff
@@ -1441,8 +1443,9 @@ impl GitPanel {
 
         let task = if self.has_staged_changes() {
             // Repository serializes all git operations, so we can just send a commit immediately
-            let commit_task =
-                active_repository.update(cx, |repo, cx| repo.commit(message.into(), None, cx));
+            let commit_task = active_repository.update(cx, |repo, cx| {
+                repo.commit(message.into(), None, self.amend_commit, cx)
```

Comment:
The amend flag does not change upon pressing commit. Do we want to reset it? Also, do we want to pre-fill the commit message with the original commit message that is being amended?